### ISSUE TITLE
CAS Service: red flags after adding SRR number 

### ISSUE THREAD

**Issue by mshukla1** [PATRIC3/patric3_website#2086]
I added SRR6691689 in the SRA accession box and clicked the arrow -> next to it and encountered the following two problems.

1. Sometimes when I click on the arrow, the form jumps a little bit and the reads don't show up in the read selection box on the right. 

2. Other times, the reads show up in the box on the right. However, Taxonomy label and output name boxes get highlighted in red. Given that I am filling out information from top to bottom and I haven't got to those fields yet, they should not be flagged until I hit submit. 

-Maulik

<img width="818" alt="screen shot 2018-08-16 at 3 46 52 pm" src="https://user-images.githubusercontent.com/1236463/44234287-de6c1d00-a16b-11e8-8fab-c9cc043ae3a9.png">

**Comment by dawenx** [comment]
From my test, I'm not able to reproduce problem 1, but this happened in the debug mode.

As for problem 2, a validation call is made before xhr request is finished, i observed the same behavior is the Assembly service. i plan to discuss with @aswarren when he is back.

**Comment by aswarren** [comment]
This issue is not unique to the SRA widget. This stems mostly from dojo form validation being called after changes to an input field. It ends up calling validation on all widgets in the form.
After talking with @dawenx today I believe the onchange:validate setup is the primary culprit:

https://github.com/PATRIC3/p3_web/blob/master/public/js/p3/widget/app/templates/ComprehensiveGenomeAnalysis.html#L1

There are some cases where we have disconnected widgets from whole form validation but I believe we are relying on this mechanism as the primary trigger. If we want to quell this problem we need to identify a design pattern/ mechanism to trigger per widget validation. @dmachi

**Comment by aswarren** [comment]
May be as simple as moving onchange:validate from the top of the template to the declaration of all widgets...

**Comment by dawenx** [comment]
I forgot to update this, but I do have a fix for reported issue.

This is more complicated then initially thought, but Andrew gave me enough help before our flight back to Blacksburg a while ago and I was able to fix it after some more tackling.

I also made change on the SRR input to provide status update as it could take a while to fetch data after user provides SRR. If a wrong SRR ID is given, the UI will remove it and give feedback right below the SRR input box.
